### PR TITLE
Baseurl fix

### LINKED ISSUE #205: Url Path
How to serve the app from another url path, e.g. ```http://localhost:3200/explorer/``` instead of ```http://localhost:3200/``` ?

### LINKED ISSUE #252: Run btc-rpc-explorer outside of the root domain
Is it possible to run btc-rpc-explorer outside of the root domain?

It looks like images/URLs are hardcoded to use the root domain. If I expose the server on a path via a proxy the images cannot be loaded and links are broken. Is there a way to support this?

e.g If I run btc-rpc-explorer at http://umbrel-dev.local everything works. But if I run it at http://umbrel-dev.local/app/btc-rpc-explorer/ no images can be loaded and all links break.

Screenshot 2020-11-20 at 19 10 40

Is there a way to set an alternate site root?

### PR DESCRIPTION
**Pull request by Kixunil**
Base URL support - fixed conflicts and config

Supercedes #207 
Closes #205 
Closes #252

### CODE REVIEW

**Comment by shesek** [comment]
Is it expected that the btc-rpc-explorer nodejs http server be the one that process the request with the subdirectory in it?

The typical use-case I have in mind is setting a reverse proxy like nginx to forward requests from `/some/subdirectory` to the btc-rpc-explorer http server. The reverse proxy would normally strip the base subdirectory and forward the request to btc-rpc-explorer, which will not see the subdirectory in the requested path.

For that case, btc-rpc-explorer should process requests normally with `/` as the root (so for example `app.use('/api/', ...)` and not `app.use(config.baseUrl + 'api/', ...)`). The only thing it needs to do differently is to use the base url for links and for loading resources.

**Comment by Kixunil** [comment]
@shesek I personally don't care, my deployments support both ways of dealing with this - it's one bool for me. I just took the code from @ketominer, merged updates and fixed two remaining bugs I knew about.

I know pretty much nothing about modern JS techniques, pug, etc... so if what you described is considered idiomatic approach and people prefer that, I'm fine with it, just don't expect me being able to change the code - someone else will have to do it. :) Just please if you decide to do it, don't take eternity. Unless there's some very serious problem with the PR, I think it'd be more useful to have this feature sooner than putting it off because the code is not perfect.

**Comment by shesek** [comment]
If it is agreed that the approach I suggested is preferable, I can go ahead and make the necessary changes.

But perhaps we should wait for feedback from @janoside and @ketominer first?

**Comment by janoside** [comment]
@shesek I agree with @Kixunil above - my goal is to support both methods. Like you, I tend to run behind an nginx proxy and personally don't plan to set my base URL to anything other than the default `/`, but I can understand the desire to run without a proxy and have node directly handle subdirectory URLs. I don't want to over-complicate things by adding too much configurability, but my intuition says both methods could have meaningful usage. That said, some changes are still needed to support the method you described.

I'm also concerned that there are some issues with the current pull request just using `./` in front of URLs because I suspect that method will fail to produce proper link URLs on pages like `/block-height/12345` with 2 path components. I'm trying to get the proposed code running in order to test, but the site just completely fails to load for me when running locally (a case where I have no proxy and just want node to handle full URLs). @Kixunil can you share any detail about your test setup?

**Comment by janoside** [comment]
Ok...sorted out some of my issues. I now see the `<base>` HTML tag so understand that the basic approach in this PR concerning relative URLs is probably good. My local instance is now (almost) working - I had 2 issues, 1 is fixed and 1 remains:

1. I didn't look closely enough at the changeset and didn't add a trailing slash on my baseUrl value in `.env` so none of the assets in `/public` were loading (that's my fault, but I'd like to make the PR more forgiving in this regard),
2. It seems that the `url(...)` values in `fonts.css` aren't working meaning the font assets fail to load and the page load just hangs (if I hardcode the proper font URLs the page loads). I'm still curious if your test setup is different in some way to where this isn't broken for you. For reference, I'm on macOS and testing in Firefox and Chrome primarily.

**Comment by Kixunil** [comment]
Oh, I just took a break from trying to remember what I changed in my setup since it stopped working for me. Looks like I forgot to update my config generator after figuring it out the first time.

1. - yes, trailing slash should be optional, I will try to do it tomorrow
2. - I can test tomorrow but I don't know how I distinguish failed load of fonts. To me it's all the same :D

**Comment by shesek** [comment]
> but I can understand the desire to run without a proxy and have node directly handle subdirectory URLs

Out of curiosity, what are the use cases you see for this for this?

File: app/coins/btc.js [hunk 1]
```diff
@@ -53,9 +53,9 @@ module.exports = {
 	name:"Bitcoin",
 	ticker:"BTC",
 	logoUrlsByNetwork:{
-		"main":"/img/logo/btc.svg",
-		"test":"/img/logo/tbtc.svg",
-		"regtest":"/img/logo/tbtc.svg"
+		"main":"./img/logo/btc.svg",
```
Comment (shesek) [inline app/coins/btc.js:56]: FYI this will work the same without the leading `./` (here and everywhere else in this PR)

Comment (Kixunil) [reply app/coins/btc.js:56]: Which way is more idiomatic? I like the explicitness of `./`, but if it's better without it, then I can remove it.

**Comment by Kixunil** [comment]
I added the fix for missing trailing slash. How can I check the fonts?

File: app/coins/btc.js (hunk 1 above)
Comment (shesek) [reply app/coins/btc.js:56]: I think that without is more idiomatic, but no big deal either way. If you like the explicitness then go for it :)

**Comment by Kixunil** [comment]
I have figured out and fixed the fonts issue. Is there anything else blocking the merge?

**Comment by janoside** [comment]
@Kixunil Well done on the fonts fix - my testing resulted in the same fix and was the last piece of the puzzle for me. Finishing up testing today and will be merging. Thanks for your persistence on this issue.

**Comment by ketominer** [comment]
Thank you guys for your help on this! I will be finally able to upgrade btc-rpc-explorer on the nodls 👍

**Comment by janoside** [comment]
@Kixunil @ketominer Changes have been merged along with a few other issues I found during testing. Thanks again for the PRs on this issue and let me know if you run into any issues with it. I'll be working toward an official release including this soon.

**Comment by Kixunil** [comment]
Awesome! Thank you too!

**Pull request merged**